shell/platform/android/android_surface_gl.cc

```diff
@@ -92,6 +92,9 @@ bool AndroidSurfaceGL::SetNativeWindow(
   FML_DCHECK(IsValid());
   FML_DCHECK(window);
   native_window_ = window;
+  // Ensure the destructor is called since it destroys the `EGLSurface` before
+  // creating a new onscreen surface.
+  onscreen_surface_ = nullptr;
```

Comment:
`onscreen_surface_.reset();` might be _very marginally_ more readable. At least to me, that way reads more as "destroy what's there" as opposed to "set to null" even though realistically the end result is the same.